swap/swap.go

```diff
@@ -184,9 +209,10 @@ func (swap *Swap) Close() {
 // resetBalance is called:
 // * for the creditor: on cheque receival
 // * for the debitor: on confirmation receival
-func (s *Swap) resetBalance(peer enode.ID) {
+func (s *Swap) resetBalance(peerID enode.ID) {
 	//TODO: reset balance based on actual amount
-	s.balances[peer] = 0
+	//Lock() is not applied because it is expected to be done in the handleChequeRequestMsg call
+	s.balances[peerID] = 0
```

Comment:
Look into reducing the balance by the cheque amount instead of setting it back to 0.

Reply:
During our discussion we concluded it should be a cumulative value and compared as such. So in practice current balance would be retrieved as such:

balance = balanceCummulative - lastChequeAmount

and the trigger for cheque request

balance >= treshHold

So this function should not exist.

Reply:
https://github.com/ethersphere/swarm/issues/1551